src/StatsdClient/DogStatsdService.cs

```diff
@@ -71,8 +72,7 @@ public void Event(string title, string text, string alertType = null, string agg
         /// <param name="value">A given delta.</param>
         /// <param name="sampleRate">Percentage of metric to be sent.</param>
         /// <param name="tags">Array of tags to be added to the data.</param>
-        /// <typeparam name="T">The type of the value.</typeparam>
-        public void Counter<T>(string statName, T value, double sampleRate = 1.0, string[] tags = null)
+        public void Counter(string statName, double value, double sampleRate = 1.0, string[] tags = null)
```

Comment:
The generic <T> to double should not break the API as an implicit cast would happen here. Correct?

Reply:
There is an implicit cast to double when no information is lost.

The problem is that `T` can be `BitInteger` https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger?view=netcore-3.1 and there is no implicit conversion to double. It can also be the string `"42"` as the previous code only assumes that there is a `ToString()` method which is available for all objects in C#.

This is a breaking change even if it is unlikely to break anything.

Reply:
Do you plan to release this in a new major?

Reply:
I am going to release this in a new major as it may break compilation in some cases.